src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtEmailServiceTest.java

```diff
@@ -0,0 +1,161 @@
+        new Email(TEST_EMAIL_ADDRESS, TEST_EMAIL_EXPLANATION, TEST_EMAIL_EXPLANATION_CY, TEST_EMAIL_TYPE),
+        new Email(TEST_EMAIL_ADDRESS2, TEST_EMAIL_EXPLANATION2, TEST_EMAIL_EXPLANATION_CY2, TEST_EMAIL_TYPE2),
+        new Email(TEST_EMAIL_ADDRESS3, TEST_EMAIL_EXPLANATION3, TEST_EMAIL_EXPLANATION_CY3, TEST_EMAIL_TYPE3)
+    );
+
+    @Autowired
+    private AdminCourtEmailService adminService;
+
+    @MockBean
+    private CourtRepository courtRepository;
+
+    @MockBean
+    private CourtEmailRepository emailRepository;
+
+    @MockBean
+    private EmailTypeRepository emailTypeRepository;
+
+    @Mock
+    private Court court;
+
+    @BeforeAll
+    static void setUp() {
+
+        for (int i = 0; i < EMAIL_COUNT; i++) {
+            CourtEmail courtEmail = mock(CourtEmail.class);
+            COURT_EMAILS.add(courtEmail);
+        }
+
+        when(COURT_EMAILS.get(0).getEmail())
+            .thenReturn(new uk.gov.hmcts.dts.fact.entity.Email(
+                TEST_EMAIL_ADDRESS, TEST_EMAIL_EXPLANATION, TEST_EMAIL_EXPLANATION_CY,
+                new uk.gov.hmcts.dts.fact.entity.EmailType(TEST_EMAIL_TYPE,
+                                                           TEST_EMAIL_DESCRIPTION, TEST_EMAIL_DESCRIPTION_CY)));
+
+        when(COURT_EMAILS.get(1).getEmail())
+            .thenReturn(new uk.gov.hmcts.dts.fact.entity.Email(
+                TEST_EMAIL_ADDRESS2, TEST_EMAIL_EXPLANATION2, TEST_EMAIL_EXPLANATION_CY2,
+                new uk.gov.hmcts.dts.fact.entity.EmailType(TEST_EMAIL_TYPE2,
+                                                           TEST_EMAIL_DESCRIPTION2, TEST_EMAIL_DESCRIPTION_CY2)));
+
+        when(COURT_EMAILS.get(2).getEmail())
+            .thenReturn(new uk.gov.hmcts.dts.fact.entity.Email(
+                TEST_EMAIL_ADDRESS3, TEST_EMAIL_EXPLANATION3, TEST_EMAIL_EXPLANATION_CY3,
+                new uk.gov.hmcts.dts.fact.entity.EmailType(TEST_EMAIL_TYPE3,
+                                                           TEST_EMAIL_DESCRIPTION3, TEST_EMAIL_DESCRIPTION_CY3)));
+    }
+
+    @Test
+    void shouldReturnAllEmails() {
+        when(court.getCourtEmails()).thenReturn(COURT_EMAILS);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+
+        assertThat(adminService.getCourtEmailsBySlug(COURT_SLUG))
+            .hasSize(EMAIL_COUNT)
+            .first()
+            .isInstanceOf(Email.class);
+    }
+
+    @Test
+    void shouldUpdateCourtOpeningTimes() {
+
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(court.getCourtEmails()).thenReturn(COURT_EMAILS);
+        when(emailRepository.saveAll(any())).thenReturn(COURT_EMAILS);
+
+        List<Email> emails = adminService.updateEmailListForCourt(COURT_SLUG, EXPECTED_EMAILS);
+        verify(emailRepository, times(1)).deleteAll(COURT_EMAILS);
+        verify(emailRepository, times(1)).saveAll(anyIterable());
```

Comment:
Think verify default to times(1). We could just use 
verify(emailRepository).deleteAll(COURT_EMAILS);
verify(emailRepository)).saveAll(anyIterable());

Reply:
changed, didint know that. Just tried it as well and youre right